Projects/Simba/framefunctionlist.pas

```diff
@@ -101,7 +101,8 @@   TFunctionListFrame = class(TFrame)
 implementation
 
 uses
-  SimbaUnit, Graphics, stringutil, simpleanalyzer, v_ideCodeParser, lclintf, dynlibs;
+  SimbaUnit, Graphics, stringutil, simpleanalyzer, v_ideCodeParser, lclintf,
+  dynlibs, LazFileUtils;
```

Comment:
LazUTF8 needs to go here, correct?

Reply:
No. The modifications compile as they are. I only added the new unit when there were errors and didn't end up adding it in some files I expected to add it to. The changes were fairly complicated and I am not exactly sure why some functions were moved.

Reply:
If we used the UTF8 functions in this file before it will need LazUTF8 now.

Just because it compiles, doesn't mean it is correct.

Reply:
Really any unit that uses non-hardcoded file access should have it.

Especially any units used to give scripts access to file handling.

Reply:
The default types of keywords changed, and the type system should be ensuring that the proper functions are being used. It is not always necessary to use special UTF8 functions and some of the Lazarus 1.8 updates made changes in this vein.

Later units mask earlier ones. Some functions in `FileUtils` will mask the functions in `LazUTF8` and `LazFileUtils`. I encountered this in a few places, I think [this is one](https://github.com/MerlijnWajer/Simba/pull/410/files/b69cb1acd34ecc42edc6ca48ede0c69879d68ff1#diff-28d2ea72f40cd81a30c2aacfa5697944R124). Some of the imports might be redundant. Separate PRs to refactor the imports might be good to ask for, as well as other tidying-up.

This is also a comment to your response below. Some units have ordering requirements, but I didn't find any between those two.